Frontend/ui-driver/src/Screens/BookingOptionsScreen/View.purs

```diff
@@ -39,33 +39,183 @@ screen initialState =
+              ][ toggleView push state.props.downgraded canDowngrade ]
+          ]
         , linearLayout
-            [ height WRAP_CONTENT
-            , width MATCH_PARENT
-            , orientation VERTICAL
-            , padding $ Padding 16 16 16 16
-            ]
-            [ defaultVehicleView push state
-            ]
-        , downgradeOptionsView push state
+          [ width MATCH_PARENT
+          , height $ V 1
+          , margin (Margin 16 0 16 16)
+          , background Color.grey700
+          ][]
         , linearLayout
-          [ height MATCH_PARENT
-          , width $ V 1
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , margin (Margin 16 0 16 16)
+          , padding (Padding 16 16 16 16)
+          , orientation HORIZONTAL
+          , background Color.linen
+          , cornerRadius 8.0
+          , visibility if canDowngrade then GONE else VISIBLE
+          ][  imageView
+              [ width $ V 15
+              , height $ V 15
+              , margin (Margin 0 3 8 0)
+              , imageWithFallback $ "ny_ic_info_orange," <> (getAssetStoreLink FunctionCall) <> "ny_ic_info_orange.png"
+              ]
+            , textView $
+              [ width MATCH_PARENT
+              , height WRAP_CONTENT
+              , text (getString DOWNGRADE_AVAILABLE_ONLY_FOR_AC_VEHICLES)
+              , color Color.black800
+              ] <> FontStyle.body1 TypoGraphy  
+           ]
+        , textView $
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , margin (Margin 16 0 16 20)
+          , text $ (getString DOWNGRADING_VEHICLE_WILL_ALLOW_YOU_TO_TAKE_BOTH_1) 
+                    <> " " <> downgradeFrom <> " "
+                    <> (getString DOWNGRADING_VEHICLE_WILL_ALLOW_YOU_TO_TAKE_BOTH_2)
+                    <> " " <> downgradeTo <> " "
+                    <> (getString DOWNGRADING_VEHICLE_WILL_ALLOW_YOU_TO_TAKE_BOTH_3)
+          , color Color.black700
+          , visibility if canDowngrade then VISIBLE else GONE
+          ] <> FontStyle.body1 TypoGraphy 
+        , linearLayout
+          [ width MATCH_PARENT
+          , height WRAP_CONTENT
+          , orientation HORIZONTAL
+          , margin (Margin 12 0 12 16)
+          , visibility if canDowngrade then VISIBLE else GONE
           , weight 1.0
-          ] []
-        , PrimaryButton.view (push <<< PrimaryButtonAC) (primaryButtonConfig state)
-        ]
+          ] $ [  downgradeVehicleCard state.data.vehicleType true
+              ] <>  ( case state.data.downgradeOptions DA.!! 0 of
+                        Just vehicle -> [downgradeVehicleCard vehicle.vehicleVariant state.props.downgraded]
+                        Nothing      -> []
+                    )
+      ]
+
+downgradeVehicleCard :: forall w. String -> Boolean -> PrestoDOM (Effect Unit) w
+downgradeVehicleCard variant enabled =
+  frameLayout
+  [ width WRAP_CONTENT
+  , height WRAP_CONTENT
+  , weight 1.0
```

Comment:
@ashishsingh101 is this weight required here?

Reply:
yes it is required